python/GafferImageUITest/CatalogueUITest.py

```diff
@@ -67,9 +67,34 @@ def testBoxedCatalogue( self ) :
 		Gaffer.PlugAlgo.promote( s["b"]["c"]["images"] )
 		Gaffer.PlugAlgo.promote( s["b"]["c"]["imageIndex"] )
 
+		# Test metadata reset data or missing due to earlier promotion
+		Gaffer.Metadata.deregisterValue( s["b"]["imageIndex"], "catalogue:columns" )
+
 		sw = GafferUI.ScriptWindow.acquire( s )
 		ne = GafferUI.NodeEditor.acquire( s["b"] )
 
+	def testInvalidColumns( self ) :
+
+		# Check we don't abort - columns may have been removed
+		# from config but still be referenced by plug metadata.
+
+		s = Gaffer.ScriptNode()
+		s["c"] = GafferImage.Catalogue()
+
+		Gaffer.Metadata.registerValue(
+			s["c"]["imageIndex"], "catalogue:columns",
+			IECore.StringVectorData( [ "Name", "Unknown" ] )
+		)
+
+		sw = GafferUI.ScriptWindow.acquire( s )
+
+		with IECore.CapturingMessageHandler() as mh :
+			ne = GafferUI.NodeEditor.acquire( s["c"] )
```

Comment:
Do we need the ScriptNode, ScriptWindow and NodeEditor here, rather than just using `NodeUI.create( catalogue )` directly? If possible, let's go with the simpler option and keep the test scoped more tightly to the bits we care about.

Reply:
Due to complications arising from lazy UI creation on 0.56_maintenance, we're going to keep this as it is for now...